packages/pyright-internal/src/analyzer/binder.ts

```diff
@@ -1438,18 +1446,17 @@ export class Binder extends ParseTreeWalker {
             const symbol = this._bindNameToScope(this._currentScope, symbolName);
             if (
                 symbol &&
+                (this._currentScope.type === ScopeType.Module || this._currentScope.type === ScopeType.Builtin) &&
                 (!node.alias ||
                     node.module.nameParts.length !== 1 ||
                     node.module.nameParts[0].value !== node.alias.value)
             ) {
-                if (this._fileInfo.isStubFile) {
+                if (this._fileInfo.isStubFile || this._fileInfo.isInPyTypedPackage) {
```

Comment:
so, now even if import is not written as "import x as x", if x is in __ all __ , it is not marked as hidden?

previously, stub didn't care about __ all __ to mark symbol as hidden or not. it only cared about 'import x as x" form.

Reply:
Yeah, the challenge I ran into is that there are symbols in some stubs that begin with `_` that are meant to be exported. The `__all__` mechanism was the only way I could think of supporting it. This came up frequently in the hand-rolled stubs that I maintain for my team's code base, but some of these `_` symbols show up even in typeshed stubs. For example, consider the function `os._exit`.

Reply:
this will impact auto-import behavior. but I guess that's fine. (since auto import decide what to include in auto import using the symbol. externallyHidden flag.

Reply:
Yeah, I consider the current behavior for auto-import to be a bug. It suggests completions that will result in runtime exceptions. This change addresses that bug.

Reply:
I just posted a question in typeshed about symbols within a type stub that begin with an underscore. https://github.com/python/typeshed/issues/6018